cmd/rpcdaemon/commands/eth_api.go

```diff
@@ -128,9 +128,10 @@ type BaseAPI struct {
 	evmCallTimeout time.Duration
 	dirs           datadir.Dirs
 	l2RpcUrl       string
+	zeroGas        bool
 }
 
-func NewBaseApi(f *rpchelper.Filters, stateCache kvcache.Cache, blockReader services.FullBlockReader, agg *libstate.AggregatorV3, singleNodeMode bool, evmCallTimeout time.Duration, engine consensus.EngineReader, dirs datadir.Dirs) *BaseAPI {
+func NewBaseApi(f *rpchelper.Filters, stateCache kvcache.Cache, blockReader services.FullBlockReader, agg *libstate.AggregatorV3, singleNodeMode bool, evmCallTimeout time.Duration, engine consensus.EngineReader, dirs datadir.Dirs, zeroGas bool) *BaseAPI {
```

Comment:
I am beginning to wonder if we should actually change the constructor here and instead have a method "SetGasless()" that will be initialized later.

Or have NewZkBaseApi with the extra param.